src/shared/components/query/GenesetsHierarchySelector.tsx

```diff
@@ -0,0 +1,112 @@
+import * as _ from 'lodash';
+import * as React from 'react';
+import * as styles_any from './styles/styles.module.scss';
+import {action, ObservableMap, expr, toJS, computed, observable} from "mobx";
+import {observer, Observer} from "mobx-react";
+import {MutSig} from "../../api/generated/CBioPortalAPIInternal";
+import ReactSelect from 'react-select';
+import GenesetsJsTree from "./GenesetsJsTree";
+import { InputHTMLAttributes, FormEvent } from "react";
+
+const styles = styles_any as {
+    GenesetsSelectorWindow: string,
+    selectionColumnHeader: string,
+    selectionColumnCell: string,
+};
+
+export interface GenesetsHierarchySelectorProps
+{
+    initialSelection: string[];
+    gsvaProfile: string;
+    sampleListId: string|undefined;
+    onSelect: (map_geneSet_selected:ObservableMap<boolean>) => void;
+}
+
+@observer
+export default class GenesetsHierarchySelector extends React.Component<GenesetsHierarchySelectorProps, {}>
+{
+    @observable appliedPercentile: string;
+    @observable appliedPvalueThreshold: string;
+    @observable appliedScoreThreshold: string;
+    @observable percentileSelectOptions: {label: string, value: string};
+    @observable pvalueThreshold: string;
+    @observable scoreThreshold: string;
+    @observable searchValue: string;
+    @observable triggerGeneSetSelection: boolean;
+    readonly percentileOptions = [{label: '50%', value: '50'}, {label: '75%', value: '75'}, {label: '100%', value: '100'}];
+    constructor(props:GenesetsHierarchySelectorProps)
+    {
+        super(props);
+        this.appliedPercentile = '75';
```

Comment:
you can actually assign these initial values at the property declarations above.  i.e. @observable appliedPercentile: string = 'xxx'.  no need to change this.  just pointing it out.